### ISSUE TITLE
Loss function spikes up

### ISSUE THREAD

**Issue by alidi8677** [google/trax#614]
### Description
Hi everyone, 

I'm training a reformer language model with a big corpus and I saw some strange behavior in my loss function. After ~18k steps the loss function jumped up to a new minimum way higher than it was before. Attached: Any ideas?
...
<img width="1032" alt="Screen Shot 2020-05-17 at 9 51 26 AM" src="https://user-images.githubusercontent.com/56310015/82154683-55908280-9824-11ea-9e72-1c19e65bc4d8.png">

### Environment information

```
OS: <your answer here>

$ pip freeze | grep tensor
# your output here

$ pip freeze | grep jax
# your output here

$ python -V
# your output here
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```

**Comment by lukaszkaiser** [comment]
This looks like a faliure of the optimization process. Which optimizer are you using? If it's Adam or a relative, maybe try a higher epsilon? Or, in general, a lower learning rate?
